File: asherah-dotnet-shared/src/Crypto/NeverExpiredCryptoPolicy.cs
using System;

namespace GoDaddy.Asherah.Crypto;

public class NeverExpiredCryptoPolicy : CryptoPolicy
{
    public override bool IsKeyExpired(DateTimeOffset keyCreationDate) => false;
    public override long GetRevokeCheckPeriodMillis() => long.MaxValue;
    public override bool CanCacheSystemKeys() => true;
    public override bool CanCacheIntermediateKeys() => true;
    public override bool CanCacheSessions() => false;
    public override long GetSessionCacheMaxSize() => long.MaxValue;
    public override long GetSessionCacheExpireMillis() => long.MaxValue;
    public override bool NotifyExpiredIntermediateKeyOnRead() => true;
    public override bool NotifyExpiredSystemKeyOnRead() => true;
    public override KeyRotationStrategy GetKeyRotationStrategy() => KeyRotationStrategy.Inline;
}

File: asherah-java/java/src/main/java/com/godaddy/asherah/crypto/NeverExpiredCryptoPolicy.java
package com.godaddy.asherah.crypto;

import java.time.Instant;

/**
 * Crypto policy where keys never expire. Maps to default native config (no expiry settings).
 * Compatible with the canonical godaddy/asherah NeverExpiredCryptoPolicy.
 */
public class NeverExpiredCryptoPolicy extends CryptoPolicy {

    @Override
    public boolean isKeyExpired(final Instant keyCreationDate) {
        return false;
    }

    @Override
    public long getRevokeCheckPeriodMillis() {
        return Long.MAX_VALUE;
    }

    @Override
    public boolean canCacheSystemKeys() {
        return true;
    }

    @Override
    public boolean canCacheIntermediateKeys() {
        return true;
    }

    @Override
    public boolean canCacheSessions() {
        return false;
    }

    @Override
    public long getSessionCacheMaxSize() {
        return Long.MAX_VALUE;
    }

    @Override
    public long getSessionCacheExpireMillis() {
        return Long.MAX_VALUE;
    }

    @Override
    public boolean notifyExpiredIntermediateKeyOnRead() {
        return false;
    }

    @Override
    public boolean notifyExpiredSystemKeyOnRead() {
        return false;
    }

    @Override
    public KeyRotationStrategy keyRotationStrategy() {
        return KeyRotationStrategy.INLINE;
    }
}
